specification/billing/resource-manager/Microsoft.Billing/stable/2021-10-01/billingSubscription.json

```diff
@@ -612,131 +681,545 @@
     },
     "BillingSubscriptionProperties": {
       "description": "The billing properties of a subscription.",
+      "type": "object",
       "properties": {
+        "autoRenew": {
+          "description": "Indicates whether auto renewal is turned on or off for a product.",
+          "type": "string",
+          "enum": [
+            "Off",
+            "On"
+          ],
+          "x-ms-enum": {
+            "name": "AutoRenew",
+            "modelAsString": true
+          }
+        },
+        "billingFrequency": {
+          "description": "The billing frequency in ISO8601 format of product in the subscription. Example: P1M, P3M, P1Y",
+          "type": "string"
+        },
+        "billingProfileId": {
+          "description": "The ID of the billing profile to which the subscription is billed. This field is only applicable for Microsoft Customer Agreement billing accounts.",
+          "type": "string"
+        },
+        "billingProfileDisplayName": {
+          "description": "The name of the billing profile to which the subscription is billed. This field is only applicable for Microsoft Customer Agreement billing accounts.",
+          "type": "string",
+          "readOnly": true
+        },
+        "billingProfileName": {
```

Comment:
billingProfileName and invoiceSectionName are included to enable retrieving subscriptions by billing profile or invoice section using a name filter.  Filtering on billingProfileId or invoiceSectionId doesn't work because those are constructed values of the full ARM paths.